conanfile.py

```diff
@@ -0,0 +1,39 @@
+from conans import ConanFile, CMake, tools
+
+
+class HelloConan(ConanFile):
+    name = "LibConfig"
+    version = "1.7.2"
+    license = "GNU LESSER GENERAL PUBLIC LICENSE"
+    author = """Mark Lindner - Lead developer & maintainer.
+                Daniel Marjamäki - Enhancements & bugfixes.
+                Andrew Tytula - Windows port.
+                Glenn Herteg - Enhancements, bugfixes, documentation corrections.s
+                Matt Renaud - Enhancements & bugfixes.
+                JoseLuis Tallon - Enhancements & bugfixes"""
+    url = "hyperrealm.github.io/libconfig/"
+    description = "Libconfig is a simple library for processing structured configuration files, like this one. This file format is more compact and more readable than XML. And unlike XML, it is type-aware, so it is not necessary to do string parsing in application code."
+    topics = ("config")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+    generators = "cmake"
+    exports_sources="*"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
```

Comment:
Create a cmake wrapper script to make sure the options (fPIC is applied + the compiler is checked).
```suggestion
        tools.save(os.path.join(self.build_folder, "CMakeLists.txt"), textwrap.dedent("""\
            cmake_minimum_required(VERSION 2.8.12)
            project(cmake_wrapper)
            include("{}/conanbuildinfo.cmake")
            conan_basic_setup()
            add_subdirectory({} libconfig}
            """).format(self.install_folder.replace("\\","/"), self.source_folder.replace("\\","/"))
        cmake = CMake(self)
        cmake.configure(source_folder=self.build_folder)
```
(untested)

Reply:
Hey, 

this code results in an infinite loop...  
why do i need this wrapper? I thougt this line: `options = {"shared": [True, False], "fPIC": [True, False]}` and the beyond one sets the options?

Reply:
Normally, it shouldn't.
The form I am aiming for is this one: https://github.com/mattgodbolt/seasocks/blob/5041aaab8d206f454c547502fdd027794f067394/conanfile.py#L53-L70

The wrapper will check the actual used compiler + apply the PIC flags (by executing `conan_basic_setup()`.

Reply:
There was a little bug in my code: the last `add_subdirectory` did not have a closing curly brace instead a closing parenthesis.

Reply:
Yeah, found that already, but it doesn't work anyway --> see upper thread.